server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNode.java

```diff
@@ -72,8 +73,13 @@ public static boolean isMasterNode(final Settings settings) {
         return hasRole(settings, DiscoveryNodeRole.MASTER_ROLE);
     }
 
+    /**
+     * Note, due to the way that plugins may not be available when settings are being initialized,
+     * do not use this from a static {@link Setting} default value function, as it may not contain
+     * all roles at that point.
```

Comment:
should we mention the `hasRole || DataTier.isExplicitDataTier` alternative in the doc here?

Reply:
`DataTier` is an x-pack concept, so I didn't want to mention it here